umap/static/umap/js/umap.features.js

```diff
@@ -483,12 +483,30 @@ L.U.FeatureMixin = {
     return false
   },
 
+  parseDateField: function (value) {
```

Comment:
Regarding your inputs here:

1. Location of the code: Actually, I was not quite sure where to put such a function, so I just put it somewhere. Maybe you could provide a small definition for the different files and examples where to to put what for which reason in the developer documentation. And yes, nothing need to be instantiated, so I moved it as you suggested.
2. Support for ISO 8601: Actually this is given. I added some documentation on the function below. While writing the documentation, I figured that this code does exactly what I need it to do, but is generally not really good. Maybe it would be better for the user to explicitly state what kind of date is contained in the feature property. Can you give some feedback here?
3. Tests: I did not familiarize myself yet with that part of you code. Is there any explanation/documentation on writing such tests? If not, I will just work through it myself and figure something out. ;)

```
// the Date() constructor can handle various inputs to create a date
// - value: epoch (unix timestamp) in milliseconds
// - dateString: ISO 8601 formatted (YYYY-MM-DDTHH:mm:ss.sssZ)
// - dateObject: JS date object
// - multiple parameters for different date fields (year, month, ...)
//
// a mix of those options shall be supported without the user having
// to specify the exact format, since umap is based on json the type
// of the feature property value can only be
// - string, number or boolean
// - object or array
// - null
//
// therefore, the following inputs shall be covered
// - epoch (unix timestamp) in milliseconds as number
// - epoch (unix timestamp) in seconds as number
// - epoch (unix timestamp) in milliseconds as string
// - epoch (unix timestamp) in seconds as string
// - date in ISO 8601 format as string
//
// this function tries to guess the format of the feature property value
// and adjust it a little before passing it to Date() constructor
//
parseDateField: function (value) {
  if (value != null && parseFloat(value).toString() === value.toString()) {
    // if the string representation of the feature property value is
    // the same with and without being parsed as a float, the value is
    // a number (either of type number or string)
    //
    // numbers are assumed to be epochs (unix timestamps) but so far
    // it is unclear whether it is in seconds, milliseconds or nanoseconds
    //
    // without user input it can never be determined with certainty, but
    // by making some assumptions and sacrificing some small date ranges,
    // it is possible to work around that
    //
    value = parseFloat(value);
    if (Math.abs(value) < 10000000000) {
      // if the absolute value of that number is smaller than 10000000000,
      // it is assumed to be in seconds and must be multiplied by 1000
      //
      value = value * 1000;
    } else if (Math.abs(value) > 10000000000000) {
      // if the absolute value of that number is bigger than 10000000000000,
      // it is assumed to be in nanoseconds and must be divided by 1000
      //
      value = value / 1000;
    }
    // in all other cases the number is assumed to be in milliseconds
  }
  // in all other cases the value is passed to the Date() constructor
  // without modification and the constructor will try to make something out of it
  //
  // this can either result in something proper (e.g. string containing dateString),
  // something wrong (e.g. boolean) or an invalid date (e.g. object, array, null,
  // string not containing dateString)
  //
  return new Date(value);
}
```

Reply:
For the tests, just have a look a how it's done on this file, and add a new `describe` for your new util:

https://github.com/umap-project/umap/blob/master/umap/static/umap/test/Util.js

Reply:
Perfect, thanks a lot. Seems pretty straight forward. I will reduce complexity of the function and then add some tests. :)

Reply:
Function became obsolete, decision made that for now only those data types that are supported by the Date() constructor are used.